### PR TITLE
feat(webpack-cli): add --no-hot flag

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
Feature

**Did you add tests for your changes?**
Yes
**If relevant, did you update the documentation?**
Yes
**Summary**
added `--no-hot` flag.

Refers #1240 #717 
Fixes #1296 

**Does this PR introduce a breaking change?**
Yes

**Other information**

### CODE REVIEW

File: test/no-hot/no-hot.test.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            '[webpack-cli] You provided both --hot and --no-hot. We will use only the last of these flags that you provided in your CLI arguments',
+        );
+        expect(stdout).toBeTruthy();
+        stat(resolve(__dirname, './bin/main.js'), (err, stats) => {
+            expect(err).toBe(null);
+            expect(stats.isFile()).toBe(true);
+            done();
+        });
+    });
+});
```
Comment (evilebottnawi) [inline test/no-hot/no-hot.test.js:29]: We should test plugins array here (Do we have `HotModuleReplacemnt` here or not), it is very bad tests where we check `stdout` and `stderr` on `toBeFalsy` and `toBeTruthy`

Comment (snitin315) [reply test/no-hot/no-hot.test.js:29]: like we check here 
https://github.com/webpack/webpack-cli/blob/e64cf82d227875402ffe3e07b2f1dac845ae1f0c/packages/webpack-cli/__tests__/AdvancedGroup.test.js#L12

Comment (evilebottnawi) [reply test/no-hot/no-hot.test.js:29]: Yes

Comment (snitin315) [reply test/no-hot/no-hot.test.js:29]: @evilebottnawi how to extract webpack options from here?

Comment (evilebottnawi) [reply test/no-hot/no-hot.test.js:29]: @snitin315 I think no way to do this, maybe we can set here `stats: verbose` and check what hot runtime is included?

File: test/no-hot/no-hot.test.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            expect(err).toBe(null);
+            expect(stats.isFile()).toBe(true);
+            done();
+        });
+        readFile(resolve(__dirname, './bin/main.js'), 'utf-8', (err, data) => {
+            expect(err).toBe(null);
+            // check for absence of special functions invoked by HMR plugin only
+            expect(data).not.toContain('/* webpack/runtime/hot module replacement */');
+            expect(data).not.toContain('function hotApply');
+            expect(data).toContain('no-hot test');
```
Comment (evilebottnawi) [inline test/no-hot/no-hot.test.js:22]: Let's avoid `function hotApply` and `no-hot test`. `webpack/runtime/hot module replacement` is enough

Comment (snitin315) [reply test/no-hot/no-hot.test.js]: @evilebottnawi updated :+1:

File: test/no-hot/no-hot.test.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+'use strict';
+const { run } = require('../utils/test-utils');
+const { stat, readFile } = require('fs');
+const { resolve } = require('path');
+describe('no-hot flag', () => {
+    it('shoul,d be successful when --no-hot is passed', (done) => {
```
Comment (jamesgeorge007) [inline test/no-hot/no-hot.test.js:6]:
```suggestion
    it('should be successful when --no-hot is passed', (done) => {
```

File: test/serve/with-custom-port/serve-custom-config.test.js [hunk 4]
```diff
@@ -53,5 +53,16 @@ describe('serve with devServer in config', () => {
             expect(stdout).toContain(`http://0.0.0.0:${port}`);
             expect(stderr).toBeFalsy();
         });
+
+        it('Passing no-hot flag works alongside other server config', async () => {
```
Comment (jamesgeorge007) [inline test/serve/with-custom-port/serve-custom-config.test.js:57]: rephrase it a bit
```suggestion
        it('works fine when no-hot flag is passed alongside other server config', async () => {
```

**Pull request merged**